presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -195,27 +198,33 @@ private static RestHighLevelClient createClient(ElasticsearchConfig config, Opti
     {
         RestClientBuilder builder = RestClient.builder(
                 new HttpHost(config.getHost(), config.getPort(), config.isTlsEnabled() ? "https" : "http"))
-                .setRequestConfigCallback(
-                        configBuilder -> configBuilder
-                                .setConnectTimeout(toIntExact(config.getConnectTimeout().toMillis()))
-                                .setSocketTimeout(toIntExact(config.getRequestTimeout().toMillis())))
                 .setMaxRetryTimeoutMillis(toIntExact(config.getMaxRetryTime().toMillis()));
 
         builder.setHttpClientConfigCallback(clientBuilder -> {
+            RequestConfig requestConfig = RequestConfig.custom()
+                    .setConnectionRequestTimeout(toIntExact(config.getRequestQueuedTimeout().toMillis()))
+                    .setConnectTimeout(toIntExact(config.getConnectTimeout().toMillis()))
+                    .setSocketTimeout(toIntExact(config.getRequestTimeout().toMillis()))
+                    .build();
+            // the client builder passed to the call-back is configured to use system properties, which makes it
+            // impossible to configure concurrency settings, so we need to build a new one from scratch
+            HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClientBuilder.create().setDefaultRequestConfig(requestConfig);
```

Comment:
I am a bit worried that we may lose some options already preset in `clientBuilder` passed as an argument to `setHttpClientConfigCallback`.
Did what is the difference between provided `clientBuilder` and `HttpAsyncClientBuilder.create()`.
Are there any differences other than `systemProperties` set to `true` vs `false`?

Reply:
I see the also call out to `.setMaxConnPerRoute(DEFAULT_MAX_CONN_PER_ROUTE).setMaxConnTotal(DEFAULT_MAX_CONN_TOTAL).`

Should we replicate it here or it does not matter?

Reply:
@losipiuk, those are being set below. I made a suggestion on how to restructure the code for better readability.

Reply:
Right - did not notice that :)